ganga/GangaLHCb/Lib/LHCbDataset/BKQuery.py

```diff
@@ -165,41 +168,34 @@ def getDataset(self):
             cmd = "getDataset('%s',%s,'%s','%s','%s','%s')" % (self.path, self.dqflag, self.type, self.startDate,
                                                                self.endDate, self.selection)
         result = get_result(cmd, 'BK query error.', credential_requirements=self.credential_requirements)
-
         logger.debug("Finished Running Command")
-
         files = []
         value = result
         if 'LFNs' in value:
             files = value['LFNs']
-        if not type(files) is list:  # i.e. a dict of LFN:Metadata
-            # if 'LFNs' in files: # i.e. a dict of LFN:Metadata
+        if not type(files) is list:
             files = list(files.keys())
 
-        logger.debug("Creating DiracFile objects")
+        logger.debug("Creating dataset")
 
-        ## Doesn't work not clear why
-        from GangaDirac.Lib.Files.DiracFile import DiracFile
-        #new_files = []
-        #def _createDiracLFN(this_file):
-        #    return DiracFile(lfn = this_file)
-        #GangaObject.__createNewList(new_files, files, _createDiracLFN)
+        if compressed:
+            import os
+            from GangaLHCb.Lib.LHCbDataset import LHCbCompressedDataset
```

Comment:
Imports should preferably be at the beginning of files. Taking away the configurable will also remove the if statement.

Reply:
I'll move the import but the if needs to be there in case the user chooses to get the full `LHCbDataset`